internal/provider/provider.go

```diff
@@ -110,19 +116,33 @@ func (p *ProwlarrProvider) Configure(ctx context.Context, req provider.Configure
 		key = data.APIKey.ValueString()
 	}
 
-	if key == "" {
-		// Error vs warning - empty value must stop execution
+	var authorization string
+	if data.Authorization.IsNull() {
+		authorization = os.Getenv("PROWLARR_AUTHORIZATION")
+	} else {
+		authorization = data.Authorization.ValueString()
+	}
+
+	if key == "" && authorization == "" {
 		resp.Diagnostics.AddError(
-			"Unable to find API key",
-			"API key cannot be an empty string",
+			"Missing Authentication Credentials",
+			"Both 'api_key' and 'authorization' are empty. You must provide either an API key or an authorization token for Prowlarr authentication.",
```

Comment:
Maybe remove the leading `.`? I don't know how errors render from this project, but the message before didn't have this.

Reply:
```suggestion
			"Both 'api_key' and 'authorization' are empty. You must provide either an API key or an authorization token for Prowlarr authentication",
```
Remove it from the end of the sentence?